platform/posix/network/include/network_utilities.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef CONNECT_H_
+#define CONNECT_H_
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Logging related header files are required to be included in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define LIBRARY_LOG_NAME and  LIBRARY_LOG_LEVEL.
+ * 3. Include the header file "logging_stack.h".
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME     "TCP"
+#endif
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+#include "network_common.h"
+
+/**
+ * @brief Information on the remote server for connection setup.
+ *
+ * May be passed to #IotNetworkInterface_t.create as `pConnectionInfo`. This
+ * structure contains commonly-used parameters, but may be replaced with an
+ * alternative.
+ */
+typedef struct NetworkServerInfo
+{
+    const char * pHostName; /**< @brief Server host name. Must be NULL-terminated. */
+    size_t hostNameLength;  /**< @brief Length associated with #NetworkServerInfo.pHostName. */
+    uint16_t port;          /**< @brief Server port in host-order. */
+} NetworkServerInfo_t;
+
+NetworkStatus_t TCP_Connect( int * pTcpSocket,
+                             const NetworkServerInfo_t * pServerInfo );
+NetworkStatus_t TCP_Disconnect( int tcpSocket );
+
+NetworkStatus_t TCP_SetSendTimeout( int tcpSocket,
+                                    uint32_t timeout );
+NetworkStatus_t TCP_SetRecvTimeout( int tcpSocket,
+                                    uint32_t timeout );
```

Comment:
This assumes that the socket descriptor must already be valid before setting a timeout. However, another option is to have a static global that tracks the timeout, and then utilitized by `TCP_Connect` to set the timeout before returning.